sdcm/cluster.py

```diff
@@ -286,6 +286,11 @@ def init(self) -> None:
         self._init_port_mapping()
 
         self.set_keep_alive()
+        if self.node_type == "db":
```

Comment:
Actually, I think it should be controlled by a yaml parameter that decide if we do this or not to avoid blockers.

Reply:
what do you mean with "avoid blockers"?
another thing, AFAIU it is possible that we won't receive reactor stalls without configuring it in all tests

Reply:
I mean that if it'll cause issues or we suspect it causing issues, we can set the default to "false" and run.

Reply:
i'm pretty confident that changing this parameter won't affect anything in the nodes, but i will confirm it

Reply:
@roydahan , @avikivity confirmed that it won't cause any issues, so no reason to have it optional (if it is not set, we might fail to get `Reactor stalls`)

Reply:
Ok, so wrap it with try;except so if the  command fail for some reason, nothing will happen to the tests.

Reply:
done

Reply:
Thought  about it again, what happens in longevities?
Do you know how does this affect the logs?
Are we going to see many of these in the logs? Do we really  care about them?

Reply:
i will run a longevity with this code + the kernel events, and see if it will respect the `blocked-reactor-notify-ms` set, or if it comes apart from that...

Reply:
i found that, while a node is starting up, these events shall be printed (right now they are failing, because the kernel is not set to dump to logs these information):
```
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 0] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 1] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 3] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 2] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 4] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 6] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 7] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 5] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 10] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 9] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 13] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 11] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 12] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 8] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
```

and right after it, we see:
```
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 0] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 1] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 5] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 7] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 9] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 12] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 3] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 8] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 10] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 4] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 11] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 13] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 6] seastar - updated: blocked-reactor-notify-ms=1000000
Nov 15 13:03:06 longevity-100gb-4h-master-db-node-8032e666-1 scylla[10819]:  [shard 2] seastar - updated: blocked-reactor-notify-ms=1000000
```

and after that, for this specific node there were no more "tries" to dump these stacks to log (so i guess it won't mess with the logs)...
anyway, i have a running job with this code + the events commit [here](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fabio/job/kernel_stack_backtrace/job/longevity-100gb-4h-kernel-events/2/)

Reply:
@roydahan , right now the test is running for ~1/2 of its runtime and we have few `kernel callstack` entries in the log, as the following:
```
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-1/system.log | wc -l
80
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-2/system.log | wc -l
0
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-3/system.log | wc -l
0
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-4/system.log | wc -l
80
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-5/system.log | wc -l
0
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$ grep "kernel callstack" longevity-100gb-4h-enable-k-db-node-00e8c9bb-6/system.log | wc -l
0
ubuntu@ip-10-0-2-152:~/sct-results/20211116-091824-634531/longevity-100gb-4h-enable-k-db-cluster-00e8c9bb$
```

and events, we have 60 entries:
```
grep DatabaseLogEvent debug.log | grep -v REACTOR_STALLED | wc -l
60
```

and an example of these events:
```
2021-11-16 09:57:08.718 <2021-11-16 09:57:04.000>: (DatabaseLogEvent Severity.DEBUG) period_type=one-time event_id=9445dcd4-4842-4de9-9c48-6bf9c1b97240: type=KERNEL_CALLSTACK regex=kernel callstack: 0x.{16} line_number=55101 node=Node longevity-100gb-4h-enable-k-db-node-00e
8c9bb-1 [54.216.99.12 | 10.0.0.129] (seed: True)
2021-11-16T09:57:04+00:00  longevity-100gb-4h-enable-k-db-node-00e8c9bb-1 !INFO    | scylla[10158]: kernel callstack: 0xffffffffffffff80 0xffffffffbb262a18 0xffffffffbad21862 0xffffffffbb1646cd 0xffffffffbb164f6d 0xffffffffbb1980b3 0xffffffffbb0afefe 0xffffffffbb0b0252 0xff
ffffffbb0b0348 0xffffffffbb0b248c 0xffffffffbb0b24ef 0xffffffffba8044c7 0xffffffffbb40008c
2021-11-16 09:57:08.725 <2021-11-16 09:57:04.000>: (DatabaseLogEvent Severity.DEBUG) period_type=one-time event_id=9445dcd4-4842-4de9-9c48-6bf9c1b97240: type=KERNEL_CALLSTACK regex=kernel callstack: 0x.{16} line_number=55102 node=Node longevity-100gb-4h-enable-k-db-node-00e
8c9bb-1 [54.216.99.12 | 10.0.0.129] (seed: True)
2021-11-16T09:57:04+00:00  longevity-100gb-4h-enable-k-db-node-00e8c9bb-1 !INFO    | scylla[10158]: kernel callstack: 0xffffffffffffff80 0xffffffffba8fe354 0xffffffffbb13016a 0xffffffffbb0d5cbb 0xffffffffbb0d6030 0xffffffffbb154ffe 0xffffffffbb15537f 0xffffffffbb1554cd 0xff
ffffffbb156d95 0xffffffffbb15645d 0xffffffffbb1567ba 0xffffffffbb179b90 0xffffffffbb173c9c 0xffffffffbb175862 0xffffffffbb176077 0xffffffffbb15f24d 0xffffffffbb164dd5 0xffffffffbb164f6d 0xffffffffbb1980b3 0xffffffffbb0afefe 0xffffffffbb0b0252 0xffffffffbb0b0348 0xffffffffbb
0b248c 0xffffffffbb0b24ef 0xffffffffba8044c7 0xffffffffbb40008c
2021-11-16 09:57:08.729 <2021-11-16 09:57:04.000>: (DatabaseLogEvent Severity.DEBUG) period_type=one-time event_id=9445dcd4-4842-4de9-9c48-6bf9c1b97240: type=KERNEL_CALLSTACK regex=kernel callstack: 0x.{16} line_number=55103 node=Node longevity-100gb-4h-enable-k-db-node-00e
8c9bb-1 [54.216.99.12 | 10.0.0.129] (seed: True)
2021-11-16T09:57:04+00:00  longevity-100gb-4h-enable-k-db-node-00e8c9bb-1 !INFO    | scylla[10158]: kernel callstack: 0xffffffffffffff80 0xffffffffbace132a 0xffffffffc05a063a 0xffffffffbb03cd43 0xffffffffbb03ced2 0xffffffffbace11ff 0xffffffffbace1498 0xffffffffbab60375 0xff
ffffffbab60667 0xffffffffbab609c5 0xffffffffbab5db05 0xffffffffbab60061 0xffffffffc0617b93 0xffffffffc0618149 0xffffffffbab35bcc 0xffffffffbab3865b 0xffffffffbab3884e 0xffffffffbab38fe0 0xffffffffba8044c7 0xffffffffbb40008c
2021-11-16 09:56:30.609 <2021-11-16 09:56:24.000>: (DatabaseLogEvent Severity.DEBUG) period_type=one-time event_id=0b530ddb-1e49-475b-8e3f-1e7f7e13c84e: type=REACTOR_STALLED regex=Reactor stalled line_number=23616 node=Node longevity-100gb-4h-enable-k-db-node-00e8c9bb-4 [34.255.205.0 | 10.0.3.78] (seed: False)
2021-11-16T09:56:24+00:00  longevity-100gb-4h-enable-k-db-node-00e8c9bb-4 !INFO    | scylla[16668]: Reactor stalled for 176 ms on shard 1. Backtrace: 0x3de8fc2 0x3de7c70 0x3de8ed0 0x7fb4ab5f1a1f 0x140db46 0x14174a4 0x14e3ac6 0x14e2476 0x148d150 0x15577cf 0x14f8104 0x147a281 0x12fb28c 0x13990cd 0x1374c51 0x406b461
```

Reply:
Ok, so add a yaml parameter to control it.